tests/js/spec/views/performance/landing.spec.jsx

```diff
@@ -191,4 +222,89 @@ describe('Performance > Landing', function() {
       })
     );
   });
+
+  it('Sets default period when navigating to trends when stats period is not set', async function() {
+    const data = initializeTrendsData({query: 'tag:value'});
+    const wrapper = mountWithTheme(
+      <PerformanceLanding
+        organization={data.organization}
+        location={data.router.location}
+      />,
+      data.routerContext
+    );
+    await tick();
+    wrapper.update();
+
+    const trendsLink = wrapper.find('[data-test-id="landing-header-trends"]').at(0);
+    trendsLink.simulate('click');
+
+    expect(browserHistory.push).toHaveBeenCalledWith(
+      expect.objectContaining({
+        query: {
+          query: 'tag:value count():>1000 transaction.duration:>0',
+          statsPeriod: '14d',
+          view: 'TRENDS',
+        },
+      })
+    );
+  });
+
+  it('Navigating to trends does not modify statsPeriod when already set', async function() {
+    const data = initializeTrendsData({
+      query: 'count():>500 transaction.duration:>10',
+      statsPeriod: '24h',
+    });
+
+    const wrapper = mountWithTheme(
+      <PerformanceLanding
+        organization={data.organization}
+        location={data.router.location}
+      />,
+      data.routerContext
+    );
+    await tick();
+    wrapper.update();
+
+    const trendsLink = wrapper.find('[data-test-id="landing-header-trends"]').at(0);
+    trendsLink.simulate('click');
+
+    expect(browserHistory.push).toHaveBeenCalledWith(
+      expect.objectContaining({
+        query: {
+          query: 'count():>500 transaction.duration:>10',
+          statsPeriod: '24h',
+          view: 'TRENDS',
+        },
+      })
+    );
+  });
+
+  it('Navigating away from trends will reset query', async function() {
+    const data = initializeTrendsData({view: FilterViews.TRENDS});
+
+    const wrapper = mountWithTheme(
+      <PerformanceLanding
+        organization={data.organization}
+        location={data.router.location}
+      />,
+      data.routerContext
+    );
+    await tick();
+    wrapper.update();
+
+    const byTransactionLink = wrapper
+      .find('[data-test-id="landing-header-all_transactions"]')
+      .at(0);
+    byTransactionLink.simulate('click');
+
+    expect(browserHistory.push).toHaveBeenNthCalledWith(
+      1,
+      expect.objectContaining({
+        query: {
+          query: '',
```

Comment:
Should we be retaining the query string contents that aren't aggregates or going to be complicated?

Reply:
It should be straightforward to accomplish using `tokenizeSearch` though I'm not sure removing all the aggregates here is the expected behaviour. Many of the aggregate conditions like `p75()` still work with no problems. I think it's only a select few that has problems like `count()` and `avg(transaction.duration)` because they're not present in the performance table. So it might be more reasonable to only remove the problematic ones.

Reply:
I'll give it a shot